### PR TITLE
Feature/itstyr 62 voters

### PR DESCRIPTION
**Pull request by tuj**
This a PR on the previous PR that has not been merged yet. 

* Adds security voters to decide if users can edit items based on which group the given item belongs to.
* Replaces Theme, ThemeCategory, Category, Question forms with manual forms thereby eliminating need for ThemeCategory and Question menu links. These are not only editable through Themes and Categories.

### CODE REVIEW

File: config/packages/easy_admin.yaml [hunk 1]
```diff
@@ -33,20 +31,9 @@ easy_admin:
...
-                    - { property: 'sort_order', label: 'entity.theme_category.sort_order' }
-                    - { property: 'category', label: 'entity.theme_category.category' }
-                    - { property: 'theme', label: 'entity.theme_category.theme' }
-                sort: ['sortOrder', 'ASC']
-                actions: []
-
         Theme:
             class: App\Entity\Theme
-            item_permission: ['ROLE_ADMIN', 'ROLE_GROUP_ADMIN']
+            item_permission: ['ROLE_SUPER_ADMIN', 'ROLE_ADMIN', 'ROLE_GROUP_ADMIN']
```
Comment (rimi-itk) [inline config/packages/easy_admin.yaml]: I'm not sure we should check for `ROLE_SUPER_ADMIN` here (and in other places).
As per [the role hierarchy](https://github.com/aakb/itstyr/blob/develop/config/packages/security.yaml#L7) `ROLE_SUPER_ADMIN` implies `SUPER_ADMIN`, and `ROLE_SUPER_ADMIN` should probably only be used if we really need to check for superpower.

File: public/form_category.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// See https://symfony.com/doc/current/form/form_collections.html
```
Comment (rimi-itk) [inline public/form_category.js]: Are we using webpack in this project or not (cf. https://github.com/aakb/itstyr/blob/develop/webpack.config.js)?

File: src/Entity/Question.php [hunk 3]
```diff
@@ -126,9 +126,12 @@ public function getSortOrder()
 
     /**
      * @param mixed $sortOrder
+     * @return \App\Entity\Question
      */
-    public function setSortOrder($sortOrder): void
+    public function setSortOrder($sortOrder): \App\Entity\Question
```
Comment (rimi-itk) [inline src/Entity/Question.php]: Use `use`!

File: src/Entity/User.php [hunk 4]
```diff
@@ -53,4 +51,24 @@ public function setGroups($groups): void
     {
         $this->groups = $groups;
     }
+
+    public function addGroup(GroupInterface $group): self
+    {
+        if (!$this->groups->contains($group)) {
+            $this->groups[] = $group;
+            $group->addGroup($this);
```
Comment (rimi-itk) [inline src/Entity/User.php]: Shouldn't this be `$group->addUser($this)`? We're in the `User` class, right?

File: src/Entity/User.php [hunk 5]
```diff
@@ -53,4 +51,24 @@ public function setGroups($groups): void
...
+        }
+
+        return $this;
+    }
+
+    public function removeGroup(GroupInterface $group): self
+    {
+        if ($this->groups->contains($group)) {
+            $this->groups->removeElement($group);
+            $group->removeGroup($this);
```
Comment (rimi-itk) [inline src/Entity/User.php]: Should this be `removeUser`?

File: src/Form/QuestionType.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class QuestionType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('sortOrder', null, [
+                'attr' => [
```
Comment (rimi-itk) [inline src/Form/QuestionType.php]: Does the form theme not handle/add this class?

File: src/Security/Voter/AbstractVoter.php [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+     */
+    protected function userHasAccessDirectAccess($subject, $token) {
+        $user = $token->getUser();
+
+        // if the user is anonymous, do not grant access
+        if (!$user instanceof User) {
+            return false;
+        }
+
+        if ($user->hasRole('ROLE_ADMIN') || $user->hasRole('ROLE_SUPER_ADMIN')) {
```
Comment (rimi-itk) [inline src/Security/Voter/AbstractVoter.php]: Maybe we need the role hierarchy here (by injection `RoleHierarchyInterface`)?

File: src/Security/Voter/ReportSystemVoter.php [hunk 8]
```diff
@@ -6,42 +6,25 @@
...
-        if ($user->hasRole('ROLE_ADMIN') || $user->hasRole('ROLE_SUPER_ADMIN')) {
+        if ($this->userHasAccessDirectAccess($subject, $token)) {
             return true;
         }
 
+        $user = $token->getUser();
         $userGroups = array_map(function($e) {
             return is_object($e) ? $e->getId() : null;
         }, $user->getGroups()->toArray());
+
```
Comment (rimi-itk) [inline src/Security/Voter/ReportSystemVoter.php]: `array_map` is not added by this PR, but `$user->getGroups()->map` should be used here.

File: src/Security/Voter/AbstractVoter.php (hunk 7 above)
Comment (rimi-itk) [reply src/Security/Voter/AbstractVoter.php]: Can`$this->authorizationChecker->isGranted` help us here?

File: translations/EasyAdminBundle.da.xlf [hunk 9]
```diff
@@ -6,6 +6,14 @@
                 <source>label.false</source>
                 <target>Nej</target>
             </trans-unit>
+            <trans-unit id="paginator.result">
```
Comment (rimi-itk) [inline translations/EasyAdminBundle.da.xlf]: Note to self: Create a PR on https://github.com/EasyCorp/EasyAdminBundle.

File: public/form_category.js (hunk 2 above)
Comment (tuj) [reply public/form_category.js]: not

File: src/Entity/User.php (hunk 4 above)
Comment (tuj) [reply src/Entity/User.php]: Bug crushed!

**Pull request merged**